src/core/SegmentedControl/SegmentedControl.stories.tsx

```diff
@@ -0,0 +1,122 @@
+      {["md", "sm", "xs"].map((size) => (
+        <div key={size} className="flex flex-col gap-4">
+          <div className="flex flex-col items-center">
+            <Badge className="mb-8">{size}</Badge>
+          </div>
+          <div
+            key={size}
+            className="flex gap-8 justify-center items-center p-8 border rounded-lg"
+          >
+            {[0, 1].map((index) => (
+              <SegmentedControl
+                key={index}
+                size={size as SegmentedControlSize}
+                active={activeIndex === index}
+                onClick={() => setActiveIndex(index)}
+                {...props}
+              >
+                Option {index + 1}
+              </SegmentedControl>
+            ))}
+          </div>
+        </div>
+      ))}
+    </div>
+  );
+};
+
+/**
+ * The default variant of the SegmentedControl component displays a grid of controls
+ * in three different sizes (md, sm, xs) with the default styling.
+ */
+export const DefaultVariant: Story = {
+  render: () => <SegmentedControlGrid />,
+};
+
+/**
+ * The subtle variant of the SegmentedControl component provides a more appropriate styling
+ * for n100/n1200 backgrounds.
+ */
+export const SubtleVariant: Story = {
+  render: () => (
+    <div className="bg-neutral-100 dark:bg-neutral-1200 rounded-lg p-16">
+      <SegmentedControlGrid variant="subtle" />
+    </div>
+  ),
+};
+
+/**
+ * The strong variant of the SegmentedControl component provides more visual contrast
+ * and is displayed against a neutral background to highlight the difference in styling.
+ */
+export const StrongVariant: Story = {
+  render: () => (
+    <div className="bg-neutral-100 dark:bg-neutral-1200 rounded-lg p-16">
+      <SegmentedControlGrid variant="strong" />
+    </div>
+  ),
+};
+
+/**
+ * This example shows the SegmentedControl with an icon positioned on the left side
+ * of the text content, demonstrating how icons can enhance visual communication.
+ */
+export const WithLeftIcon: Story = {
+  render: () => <SegmentedControlGrid leftIcon="icon-tech-javascript" />,
+};
+
+/**
+ * This example shows the SegmentedControl with an icon positioned on the right side
+ * of the text content, useful for indicating actions or additional information.
+ */
+export const WithRightIcon: Story = {
+  render: () => <SegmentedControlGrid rightIcon="icon-tech-javascript" />,
+};
+
+/**
+ * The disabled state of the SegmentedControl shows how the component appears when
+ * it's not interactive, with muted colors and a not-allowed cursor.
+ */
+export const Disabled: Story = {
+  render: () => <SegmentedControlGrid disabled />,
+};
+
+/**
+ * This variant shows the SegmentedControl with fully rounded corners (pill shape),
+ * which can be useful for certain design aesthetics or to visually distinguish
+ * different types of controls.
+ */
+export const Rounded: Story = {
+  render: () => <SegmentedControlGrid rounded />,
+};
```

Comment:
[nitpick] Not sure if an example using the `className` prop would be useful or not